src/main/scala/thurloe/database/ThurloeDatabaseConnector.scala

```diff
@@ -1,39 +1,54 @@
 package thurloe.database
 
 import com.typesafe.config.ConfigFactory
+import thurloe.crypto.{EncryptedBytes, SecretKey, Aes256Cbc}
 import scala.concurrent.{Future, Await}
 import scala.concurrent.ExecutionContext.Implicits.global
 import thurloe.service.{UserKeyValuePairs, UserKeyValuePair, KeyValuePair}
 
 import scala.concurrent.duration.Duration
+import org.apache.commons.codec.binary.{Base64, Hex}
+
 import scala.util.{Failure, Success, Try}
 
 case object ThurloeDatabaseConnector extends DataAccess {
 
-  val config = ConfigFactory.load().getConfig("database")
-  val databaseConfig = config.getConfig(config.getString("config"))
+  val configFile = ConfigFactory.load()
+
+  // DB Config:
+  val dbConfig = configFile.getConfig("database")
+  val databaseInstanceConfig = dbConfig.getConfig(dbConfig.getString("config"))
+  val dataModels = new DatabaseDataModels(databaseInstanceConfig.getString("slick.driver"))
 
-  val dataModels = new DatabaseDataModels(databaseConfig.getString("slick.driver"))
+  // Crypto Config:
+  val cryptoConfig = configFile.getConfig("crypto")
+  val secretKey = SecretKey(cryptoConfig.getString("key"))
 
   import dataModels._
   import dataModels.driver.api._
 
-  val database = Database.forConfig("", databaseConfig)
+  val database = Database.forConfig("", databaseInstanceConfig)
 
-  if (databaseConfig.hasPath("slick.createSchema") && databaseConfig.getBoolean("slick.createSchema"))
+  if (databaseInstanceConfig.hasPath("slick.createSchema") && databaseInstanceConfig.getBoolean("slick.createSchema"))
     setupInMemoryDatabase(database)
 
+  def databaseValuesToKeyValuePair(key: String, value: String, iv: String): Try[KeyValuePair] = {
+    Aes256Cbc.decrypt(EncryptedBytes(value, iv), secretKey) map { decryptedBytes =>
+      KeyValuePair(key, new String(decryptedBytes, "UTF-8"))
+    }
+  }
+
   private def lookupWithConstraint(constraint: DbKeyValuePair => Rep[Boolean]): Future[Seq[KeyValuePair]] = {
     val keyValuePairs = TableQuery[DbKeyValuePair]
     val query = keyValuePairs.filter(constraint)
 
-    val futureResults = database.run(query.result.transactionally) map {
-      _ map {
-      case (id, userId, key, value) =>
-        KeyValuePair(key, value)
-    }}
-
-    futureResults
+    for {
+      resultSequence <- database.run(query.result.transactionally)
+      tryResult = resultSequence map {
```

Comment:
IMO the right hand side of this is a bit heavyweight for the middle of a for comp